src/Core/AdminConsole/OrganizationFeatures/Policies/Implementations/SavePolicyCommand.cs

```diff
@@ -144,12 +158,168 @@ private async Task RunValidatorAsync(IPolicyValidator validator, PolicyUpdate po
         await validator.OnSaveSideEffectsAsync(policyUpdate, currentPolicy);
     }
 
-    private async Task<(Dictionary<PolicyType, Policy> savedPoliciesDict, Policy? currentPolicy)> GetCurrentPolicyStateAsync(PolicyUpdate policyUpdate)
+    private async Task<(Dictionary<PolicyType, Policy> savedPoliciesDict, Policy? currentPolicy)>
+        GetCurrentPolicyStateAsync(PolicyUpdate policyUpdate)
     {
         var savedPolicies = await _policyRepository.GetManyByOrganizationIdAsync(policyUpdate.OrganizationId);
         // Note: policies may be missing from this dict if they have never been enabled
         var savedPoliciesDict = savedPolicies.ToDictionary(p => p.Type);
         var currentPolicy = savedPoliciesDict.GetValueOrDefault(policyUpdate.Type);
         return (savedPoliciesDict, currentPolicy);
     }
+
+    public async Task<Policy> V3SaveAsync(SavePolicyModel policyModel)
```

Comment:
Here is the new `Save` method implementation. It’s very similar to the existing `Save` method, except:  

- It breaks the validations into separate methods for different types of validations.  
- It includes pre- and post-side-effect methods.  
- All these methods are on the same level, making it easy for their output to be passed to the next one.